### DISCUSSION TITLE
When using Pact server, the response message is flushed unexpectedly?

### DISCUSSION THREAD

**Discussion by CarmenRietdijk** [tokio-rs/tokio#6289] [Q&A]
Dear community,

I am a test automation engineer. I am currently working on contract testing using Pact. 
I am familiar with Java, and currently work in TypeScript. Rust is completely new to me, and so is Tokio. 
So I hope you'll bear with me in case I ask very n00b questions. 

When I try to run my test I run unto this error. From what I can tell it seems that the response message gets flushed, triggering the shutdown of the server. See the image below for more detail. 

Am I correct in my analysis? And if so, what could be the cause for the flush of the response message?

![image](https://github.com/tokio-rs/tokio/assets/95585889/7c94551c-19f1-4175-ab87-5212b7d6129a)

Thank you for your time,
Carmen

**Comment by Darksonn** [comment]
It looks like you are manually spawning threads and then try to use Tokio resources from within the custom thread.

However, new threads are outside of the Tokio context by default. You may have to use [`enter`](https://docs.rs/tokio/latest/tokio/runtime/struct.Handle.html#method.enter) on the new thread to explicitly enter the Tokio context.

**Reply by CarmenRietdijk** [reply]
Thank you for your quick reply :-) 
I will try to use this information to solve the problem.
